Src/VimCore/CommandUtil.fs

```diff
@@ -2939,11 +2948,68 @@ type internal CommandUtil
     member x.RunInsertCommand command =
         _insertUtil.RunInsertCommand command
 
+    /// Whether we should run the specified command for all carets
+    member x.ShouldRunNormalCommandForEachCaret command =

```

Comment:
Think this could be a local function within `RunNormalCommand`

Reply:
Ideally we would like to "ask the command" whether it wants to run for each caret, rather than "telling the command" we are going to run it for each caret. in other words, it should be conceptually like flag associated with the abstract normal command. I think this should therefore be moved up into `NormalCommand`.

Reply:
That makes sense